component/src/main/java/org/ballerinalang/sap/utils/SapUtils.java

```diff
@@ -260,53 +254,56 @@ public static Properties getDestinationProperties(Struct destinationConfig) {
      * @param sapService that contains resource map
      * @param error      {@link BError} instance which contains the error details
      */
-    public static void invokeOnError(Map<String, Resource> sapService, CallableUnitCallback callback, String error) {
+    public static void invokeOnError(Map<String, Resource> sapService, CallableUnitCallback callback, String error,
+                                     Context context) {
         try {
             Resource errorResource = sapService.get(RESOURCE_ON_ERROR);
-            log.info("invokeOnError : " + sapService.get(RESOURCE_ON_ERROR).getName());
-            BValue[] params = getOnErrorResourceSignature(errorResource, error);
-            Executor.submit(errorResource, callback, null, null, params);
+            if (log.isDebugEnabled()) {
+                log.info("invokeOnError : " + sapService.get(RESOURCE_ON_ERROR).getName());
+            }
+            BValue[] bValues = new BValue[1];
+            bValues[0] = createError(context, error);
+            Executor.submit(errorResource, callback, null, null, bValues);
```

Comment:
Can't we just pass `createError(context, error)` to the rest param instead of creating a new array?